py-polars/polars/io.py

```diff
@@ -704,6 +705,26 @@ def scan_parquet(
     )
 
 
+def scan_json(
+    file: str | Path | IOBase,
+) -> LazyFrame:
+    """
+    Lazily read from a JSON file.
+
+    Parameters
+    ----------
+    file
+        Path to a file.
+
+    """
+    if isinstance(file, StringIO):
+        file = BytesIO(file.getvalue().encode())
+    elif isinstance(file, (str, Path)):
+        file = format_path(file)
+
+    return LazyFrame._scan_json(file)
+
+
```

Comment:
i think this method should be renamed to something with `deserialize` in it, or another way to easily identify that this is not meant to read json data, but a json representation of a logicalplan. 

within the node bindings, i went with the signature of `Lazyframe.deserialize(data, format='json')`

Reply:
Maybe I am not understanding this correctly, but as I understood it the status quo is:

* `read_x` -> Read from file eagerly
* `scan_x` -> Read from file lazily
* `from_x` -> Create this object from another object
* `to_x` -> Create another object from this object

In your proposal, you propose to:

* Consistently handle `json` and `ndjson` separately
* Rename `from_x` -> `deserialize(format="x")`
* Rename `to_x` -> `serialize(format="x")`

The function that you commented on reads lazily from disk, so it should be called `scan`. In this PR so far, I am trying to address your first point for the JSON Input functions only.

There is still a classmethod `LazyFrame.from_json`, in your proposal this would be renamed to `LazyFrame.serialize(format="json")`. I am not planning to incorporate that into this PR.

Is this correct? If not, could you point out where I go wrong?

Reply:
We don't actually have a true `scan_json`. Scanning is only possible with newline delimited json. The current [LazyFrame.read_json](https://github.com/pola-rs/polars/blob/master/py-polars/src/lazy/dataframe.rs#L123) is the equivalent of the `from_x` action.

As you mentioned in #4395 I think the `from_x` pattern makes a lot of sense. With that in mind, this method should just be called `from_json`

Reply:
> We don't actually have a true `scan_json`. Scanning is only possible with newline delimited json. The current [LazyFrame.read_json](https://github.com/pola-rs/polars/blob/master/py-polars/src/lazy/dataframe.rs#L123) is the equivalent of the `from_x` action.

Ah, I didn't know this. Thanks for explaining this. I will adjust my PR accordingly.